cumulus_library/studies/core/builder_core_medication.py

```diff
@@ -29,60 +30,45 @@ def _check_data_in_fields(self, cursor, schema: str):
         with get_progress_bar(transient=True) as progress:
             task = progress.add_task(
                 "Detecting available medication sources...",
-                total=8,
+                total=9,
             )
 
             # inline medications from FHIR medication
-            query = get_is_table_not_empty_query(table, base_col)
-            cursor.execute(query)
-            progress.advance(task)
-            if cursor.fetchone() is not None:
+            data_types["inline"] = is_codeable_concept_populated(
+                schema, table, base_col, cursor, progress, task
+            )
+            if data_types["inline"]:
                 query = get_column_datatype_query(schema, table, base_col)
                 cursor.execute(query)
                 progress.advance(task)
-                if "coding" in str(cursor.fetchone()[0]):
-                    query = get_is_table_not_empty_query(
-                        table,
-                        "t1.row1",
-                        [
-                            {
-                                "source_col": f"{base_col}.coding",
-                                "table_alias": "t1",
-                                "row_alias": "row1",
-                            }
-                        ],
-                    )
-                    cursor.execute(query)
-                    progress.advance(task)
-                    if cursor.fetchone() is not None:
-                        data_types["inline"] = True
+                if "userselected" not in str(cursor.fetchone()[0]):
+                    missing_userselected = True
```

Comment:
nit: flip this meaning to `has_userselected` - positive variables are way easier to reason about, I'd say